### PR TITLE
Adds new event to the Orion Trail Game - Old Ship

### PR DESCRIPTION
**Pull request by Quartich**
## About The Pull Request
Adds a new event to the Orion trail arcade game - Old ship. The old ship is a medium rarity event where the player has an option to search the ship, or leave it. Searching the ship can take or give resources and crew members.

## Why It's Good For The Game
It updates content where there hasn't been any content additions in a long time. It makes an option to the player to gamble, which the other events do less so.

## Changelog
:cl: Quartich
add: Adds a new event to the Orion Trail - Old Ship
/:cl:

### CODE REVIEW

File: code/game/machinery/computer/arcade.dm [hunk 1]
```diff
@@ -800,6 +807,45 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
 			eventdat += "<P ALIGN=Right><a href='byond://?src=[REF(src)];slow=1'>Slow Down</a> <a href='byond://?src=[REF(src)];keepspeed=1'>Continue</a></P>"
 			eventdat += "<P ALIGN=Right><a href='byond://?src=[REF(src)];close=1'>Close</a></P>"
 
+		if(ORION_TRAIL_OLDSHIP)
+			eventdat += "<br>Your crew spots an old ship floating through space. It might have some supplies, but then again it looks rather unsafe."
+			eventdat += "<P ALIGN=Right><a href='byond://?src=[REF(src)];search=1'>Search it</a><a href='byond://?src=[REF(src)];eventclose=1'>Leave it</a></P><P ALIGN=Right><a href='byond://?src=[REF(src)];close=1'>Close</a></P>"
+
+		if(ORION_TRAIL_SEARCH)
+			if(prob(20))
```
Comment (nemvar) [inline code/game/machinery/computer/arcade.dm]: Wouldn't a switch make more sense here than an if else chain?

File: code/game/machinery/computer/arcade.dm [hunk 2]
```diff
@@ -800,6 +807,45 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
...
+				var/oldfood = rand(1,7)
+				var/oldfuel = rand(4,10)
+				food += oldfood
+				fuel += oldfuel
+				eventdat += "<br>As you look through it you find some supplies and a living person!"
+				eventdat += "<br>[rescued] was rescued from the abandoned ship!"
+				eventdat += "<br>You found [oldfood] <b>Food</b> and [oldfuel] <b>Fuel</b>."
+				eventdat += "<P ALIGN=Right><a href='byond://?src=[REF(src)];eventclose=1'>Continue</a></P>"
+				eventdat += "<P ALIGN=Right><a href='byond://?src=[REF(src)];close=1'>Close</a></P>"
+				canContinueEvent = 1
```
Comment (nemvar) [inline code/game/machinery/computer/arcade.dm]: It looks like these three lines are copied at the end of every outcome. You could just move them down on tab and get rid of 12 line of duplicate code.

**Comment by nemvar** [comment]
I care a great deal about orion trial balance. How about a random chance for giant spiders.

File: code/game/machinery/computer/arcade.dm [hunk 3]
```diff
@@ -592,6 +595,10 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
 			food = 80
 			fuel = 60
 			settlers = list("Harry","Larry","Bob")
+	else if(href_list["search"]) //search old ship
+		if(event == ORION_TRAIL_OLDSHIP)
+			event = "Old Ship Search"
```
Comment (nemvar) [inline code/game/machinery/computer/arcade.dm]: Replace this with
```suggestion
			event = ORION_TRAIL_SEARCH

File: code/game/machinery/computer/arcade.dm [hunk 4]
```diff
@@ -369,7 +371,8 @@ GLOBAL_LIST_INIT(arcade_prize_pool, list(
 						   ORION_TRAIL_LING			= 3,
 						   ORION_TRAIL_MALFUNCTION	= 2,
 						   ORION_TRAIL_COLLISION	= 1,
-						   ORION_TRAIL_SPACEPORT	= 2
+						   ORION_TRAIL_SPACEPORT	= 2,
+						   ORION_TRAIL_OLDSHIP		= 3
```
Comment (Quartich) [inline code/game/machinery/computer/arcade.dm]:
```suggestion
						   ORION_TRAIL_OLDSHIP		= 2
```

**Review by Quartich — COMMENTED** [review]
Forgot to change this back after testing

**Comment by Quartich** [comment]
clicked wrong button and closed

**Comment by ATH1909** [comment]
> I care a great deal about orion trial balance. How about a random chance for giant spiders.

Giant spiders are no joke.

Did you add emagged versions of the effects from this event?

**Comment by Quartich** [comment]
@ATH1909 I was considering that but have yet to do it

**Comment by ATH1909** [comment]
> @ATH1909 I was considering that but have yet to do it

Maybe make it someone near the machine (not necessarily you, although it has a chance to teleport you) to a place with safe atmos in KSS13 (the place where those drones spawn) if you roll the option where you lose a crew member in the abandoned space station? Maybe give the teleported person a box containing a space suit, an oxygen mask, and an oxygen tank as well, so they don't get stuck?

If you implement the giant spider option, you could make the emagged version of that result spawn some giant spiders.

**Comment by Quartich** [comment]
@ATH1909  Yeah I was thinking about the Giant spider for emagged, but I like that idea of teleporting someone to the derelict.

**Pull request merged**